src/main/java/com/cleanroommc/groovyscript/compat/inworldcrafting/FluidToItem.java

```diff
@@ -0,0 +1,136 @@
+        if (IngredientHelper.isEmpty(fluid)) {
+            GroovyLog.msg("Error removing in world fluid to item recipe")
+                    .add("input fluid must not be empty")
+                    .error()
+                    .post();
+            return false;
+        }
+        if (!FluidRecipe.removeIf(fluid.getFluid(), fluidRecipe -> fluidRecipe.getClass() == Recipe.class, fluidRecipe -> addBackup((Recipe) fluidRecipe))) {
+            GroovyLog.msg("Error removing in world fluid to item recipe")
+                    .add("no recipes found for {}", fluid.getFluid().getName())
+                    .error()
+                    .post();
+            return false;
+        }
+        return true;
+    }
+
+    public boolean removeByInput(FluidStack fluid, ItemStack... input) {
+        if (GroovyLog.msg("Error removing in world fluid to item recipe")
+                .add(IngredientHelper.isEmpty(fluid), () -> "input fluid must not be empty")
+                .add(IngredientHelper.isEmpty(input), () -> "input ingredients must not be empty")
+                .error()
+                .postIfNotEmpty()) {
+            return false;
+        }
+        if (!FluidRecipe.removeIf(fluid.getFluid(), fluidRecipe -> fluidRecipe.getClass() == Recipe.class && fluidRecipe.matches(input), fluidRecipe -> addBackup((Recipe) fluidRecipe))) {
+            GroovyLog.msg("Error removing in world fluid to item recipe")
+                    .add("no recipes found for {}", fluid.getFluid().getName())
+                    .error()
+                    .post();
+            return false;
+        }
+        return true;
+    }
+
+    public boolean removeAll() {
+        return FluidRecipe.removeIf(fluidRecipe -> fluidRecipe.getClass() == Recipe.class, fluidRecipe -> addBackup((Recipe) fluidRecipe));
+    }
+
+    public RecipeBuilder recipeBuilder() {
+        return new RecipeBuilder();
+    }
+
+    public SimpleObjectStream<Recipe> streamRecipes() {
+        return new SimpleObjectStream<>(FluidRecipe.findRecipesOfType(Recipe.class)).setRemover(this::remove);
+    }
+
+    public static class Recipe extends FluidRecipe {
+
+        private final ItemStack output;
+
+        public Recipe(Fluid input, IIngredient[] itemInputs, float[] itemConsumeChance, Closure<Boolean> beforeRecipe, Closure<?> afterRecipe, ItemStack output) {
+            super(input, itemInputs, itemConsumeChance, beforeRecipe, afterRecipe);
+            this.output = output;
+        }
+
+        public ItemStack getOutput() {
+            return output;
+        }
+
+        @Override
+        public void setJeiOutput(IIngredients ingredients) {
+            ingredients.setOutput(VanillaTypes.ITEM, getOutput());
+        }
+
+        @Override
+        public void handleRecipeResult(World world, BlockPos pos) {
+            world.setBlockToAir(pos);
+            world.spawnEntity(new EntityItem(world, pos.getX(), pos.getY(), pos.getZ(), getOutput().copy()));
```

Comment:
Might be better to `+ 0.5D` on `getX()` `getY()` `getZ()` so it spawns on the center of the block.

Or mimic `Block::spawnAsEntity` method where `Random` is used so it spawns the entity near the center of the block but in slightly different positions each time.